query_optimizer/tests/DistributedExecutionGeneratorTestRunner.cpp

```diff
@@ -158,5 +193,44 @@ void DistributedExecutionGeneratorTestRunner::runTestCase(
   }
 }
 
+block_id_domain DistributedExecutionGeneratorTestRunner::getBlockDomain(
+    const string &network_address) {
+  serialization::BlockDomainRegistrationMessage proto;
+  proto.set_domain_network_address(network_address);
+
+  const int proto_length = proto.ByteSize();
+  char *proto_bytes = static_cast<char*>(malloc(proto_length));
+  CHECK(proto.SerializeToArray(proto_bytes, proto_length));
+
+  TaggedMessage message(static_cast<const void*>(proto_bytes),
+                        proto_length,
+                        kBlockDomainRegistrationMessage);
+  free(proto_bytes);
+
+  LOG(INFO) << "Client (id '" << cli_id_
+            << "') sent BlockDomainRegistrationMessage (typed '" << kBlockDomainRegistrationMessage
+            << "') to BlockLocator (id '" << locator_client_id_ << "')";
+
+  CHECK(MessageBus::SendStatus::kOK ==
+      QueryExecutionUtil::SendTMBMessage(&bus_,
+                                         cli_id_,
+                                         locator_client_id_,
+                                         move(message)));
+
+  const tmb::AnnotatedMessage annotated_message(bus_.Receive(cli_id_, 0, true));
```

Comment:
From the PR description: 

> ... this PR uses BlockLocator and DataExchangerAsync, to locate all the blocks, and fetch a remote block in an async manner, respectively.

However the line above indicates that the lookup for the block happens in a synchronous manner. Am I missing something?